src/test/java/seedu/address/logic/commands/WorkerAvailableCommandTest.java

```diff
@@ -0,0 +1,92 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_ROLE_CASHIER;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_ROLE_CHEF;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.showShiftAtIndex;
+import static seedu.address.testutil.AddressBookBuilder.getTypicalAddressBook;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_SHIFT;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_SHIFT;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.tag.Role;
+
+/**
+ * Contains integration tests (interaction with the Model)
+ * and unit tests for WorkerAvailableCommand.
+ */
+public class WorkerAvailableCommandTest {
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_invalidShiftIndexUnfilteredList_failure() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredShiftList().size() + 1);
+        Role role = Role.createRole(VALID_ROLE_CASHIER);
+        WorkerAvailableCommand workerAvailableCommand = new WorkerAvailableCommand(outOfBoundIndex, role);
+
+        assertCommandFailure(workerAvailableCommand, model, Messages.MESSAGE_INVALID_SHIFT_DISPLAYED_INDEX);
+    }
+
+    /**
+     * Edit filtered list where index is larger than size of filtered list,
+     * but smaller than size of address book
+     */
+    @Test
+    public void execute_invalidShiftIndexFilteredList_failure() {
+        showShiftAtIndex(model, INDEX_FIRST_SHIFT);
+        Index outOfBoundIndex = INDEX_SECOND_SHIFT;
+        // ensures that outOfBoundIndex is still in bounds of McScheduler shift list
+        assertTrue(outOfBoundIndex.getZeroBased() < model.getAddressBook().getShiftList().size());
+
+        WorkerAvailableCommand workerAvailableCommand = new WorkerAvailableCommand(outOfBoundIndex,
+                Role.createRole(VALID_ROLE_CASHIER));
+
+        assertCommandFailure(workerAvailableCommand, model, Messages.MESSAGE_INVALID_SHIFT_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_roleNotFound_throwsCommandException() {
+        WorkerAvailableCommand workerAvailableCommand = new WorkerAvailableCommand(INDEX_FIRST_SHIFT,
+                Role.createRole("random role"));
```

Comment:
Can use CommandTestUtil.NOT_FOUND_ROLE_DESC to be consistent but probably not necessary